src/internal/connector/exchange/api/shared.go

```diff
@@ -56,6 +59,43 @@ func toValues[T any](a any) ([]getIDAndAddtler, error) {
 	return r, nil
 }
 
+func GetAddedAndRemovedItemIDsFromPager(
+	ctx context.Context,
+	oldDelta string,
+	pgr itemPager,
+) ([]string, []string, DeltaUpdate, error) {
+	added, removed, deltaURL, err := getItemsAddedAndRemovedFromContainer(ctx, pgr)
+	if err == nil {
+		return added, removed, DeltaUpdate{deltaURL, false}, nil
+	}
+
+	// if we fail with quota exceeded error, retry with non delta
+	if graph.IsErrQuotaExceeded(err) {
+		pgr.reset(true)
+
+		added, removed, deltaURL, err = getItemsAddedAndRemovedFromContainer(ctx, pgr)
+		if err == nil {
+			// TODO(meain): Should we return resetDelta here? Will
```

Comment:
@ashmrtn Wouldn't just returning `resetDelta` here take care of "non incremental" backup?

Reply:
with the way the code is arranged, yes that should work